tests/Unit/Adapter/Product/Update/Filler/SeoFillerTest.php

```diff
@@ -0,0 +1,267 @@
+            ],
+            $expectedProduct,
+        ];
+
+        $product = $this->mockDefaultProduct();
+        $command = $this->getEmptyCommand()
+            ->setRedirectOption(RedirectType::TYPE_CATEGORY_TEMPORARY, 1)
+        ;
+        $expectedProduct = $this->mockDefaultProduct();
+        $expectedProduct->redirect_type = RedirectType::TYPE_CATEGORY_TEMPORARY;
+        $expectedProduct->id_type_redirected = 1;
+
+        yield [
+            $product,
+            $command,
+            [
+                'redirect_type',
+                'id_type_redirected',
+            ],
+            $expectedProduct,
+        ];
+    }
+
+    public function testThrowsExceptionWhenNonExistingRedirectTargetIsProvided(): void
+    {
+        $seoFiller = $this->getFiller(false, false);
+
+        $this->expectException(ProductNotFoundException::class);
+        $product = $this->mockDefaultProduct();
+        $command = $this->getEmptyCommand()
+            ->setRedirectOption(RedirectType::TYPE_PRODUCT_TEMPORARY, 1)
+        ;
+        $seoFiller->fillUpdatableProperties($product, $command);
+
+        $this->expectException(CategoryNotFoundException::class);
+        $product = $this->mockDefaultProduct();
+        $command = $this->getEmptyCommand()
+            ->setRedirectOption(RedirectType::TYPE_CATEGORY_PERMANENT, 1)
+        ;
+        $seoFiller->fillUpdatableProperties($product, $command);
+    }
+
+    /**
+     * @param bool $redirectProductExists
+     * @param bool $redirectCategoryExists
+     *
+     * @return SeoFiller
+     */
+    private function getFiller(bool $redirectProductExists, bool $redirectCategoryExists): SeoFiller
+    {
+        return new SeoFiller(
+            $this->mockProductRepository($redirectProductExists),
+            $this->mockCategoryRepository($redirectCategoryExists),
+            $this->mockTools()
+        );
+    }
+
+    /**
+     * @param bool $redirectProductExists
+     *
+     * @return ProductRepository
+     */
+    private function mockProductRepository(bool $redirectProductExists): ProductRepository
+    {
+        $mock = $this->createMock(ProductRepository::class);
+        if (!$redirectProductExists) {
+            $mock->method('assertProductExists')
+                ->willThrowException(new ProductNotFoundException())
+            ;
+        }
+
+        return $mock;
+    }
+
+    /**
+     * @param bool $redirectCategoryExists
+     *
+     * @return CategoryRepository
+     */
+    private function mockCategoryRepository(bool $redirectCategoryExists): CategoryRepository
+    {
+        $mock = $this->createMock(CategoryRepository::class);
+        if (!$redirectCategoryExists) {
+            $mock->method('assertCategoryExists')
+                // we don't assert the internals of assertCategoryExists method, so categoryId and exception string is not important
+                ->willThrowException(new CategoryNotFoundException(new CategoryId(1)));
```

Comment:
This actually made me realize that the implementation of `CategoryRepository::get` is incorrect, if the category is not found there will be an error because the exception can't be built correctly It expects a `CategoryId` first parameter but the abstract class uses a string message (not the scope of this PR but it should be fixed on 8.0.x)

Reply:
yes indeed!

Reply:
Fix here https://github.com/PrestaShop/PrestaShop/pull/30160